grails-app/controllers/org/pih/warehouse/picklist/PicklistController.groovy

```diff
@@ -218,22 +219,46 @@ class PicklistController {
         }
     }
 
-    def importPickListItems(ImportDataCommand command) {
-        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
-        List<PickPageItem> pickPageItems = stockMovementService.getPickPageItems(params.id, null, null)
+    def importPickListItems(PicklistImportDataCommand command) {
+
+        // Bind stock movement based on provided ID
+        command.stockMovement = stockMovementService.getStockMovement(params.id)
+
+        // Bind user-provided location or current location
+        command.location = command.location ?: Location.get(session.warehouse.id)
 
-        MultipartFile importFile = command.importFile
-        if (importFile.isEmpty()) {
-            throw new IllegalArgumentException("File cannot be empty")
+        // Bind the parent requisition items to the command class
+        command.pickPageItems = stockMovementService.getPickPageItems(params.id, null, null)
+
+        // TODO Test that this actually captures all of the validation errors on the input file.
+        //  We could also postpone this until the
+        // Validate provided properties of the data import, including the import file
+        if(!command.validate()) {
+            throw new ValidationException("Import failed due to errors", command.errors)
         }
 
-        String csv = new String(importFile.bytes)
-        List<ImportPickCommand> importedLines = stockMovementService.parsePickCsvTemplateImport(csv)
+//        MultipartFile importFile = command.importFile
+//        if (importFile.isEmpty()) {
+//            throw new IllegalArgumentException("File cannot be empty")
+//        }
+
+//        String csv = new String(importFile.bytes)
```

Comment:
I assume this will be deleted later because you are still working on that

Reply:
Yes, but thanks for the reminder. I just wanted to make sure I tested an invalid file before removing it.